packages/patternfly-3/patternfly-react/src/components/LoadingState/LoadingState.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Spinner } from '../Spinner';
+
+class LoadingState extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      render: false
+    };
+  }
+
+  componentDidMount() {
+    this.onTimeout = setTimeout(() => {
+      this.setState({ render: true });
+    }, this.props.timeout);
+  }
+
+  componentWillUnmout() {
+    clearTimeout(this.onTimeout);
+  }
+
+  render() {
+    const { loading, loadingText, children, size, additionalClasses } = this.props;
+
+    const spinner = (
+      <div className={`loading-state-pf loading-state-pf-${size} ${additionalClasses}`}>
+        <Spinner loading={loading} size={size} />
+        {loadingText}
+      </div>
+    );
+
+    if (loading) {
+      return this.state.render ? spinner : null;
+    }
+    return children;
+  }
+}
+
+LoadingState.propTypes = {
+  /** determines whether to show spinner or children */
+  loading: PropTypes.bool,
+  /** text to show with spinner */
+  loadingText: PropTypes.string,
+  /** children nodes */
+  children: PropTypes.node,
+  /** delay in showing the children */
+  timeout: PropTypes.number,
+  /** size of the spinner */
+  size: PropTypes.oneOf(['lg', 'sm', 'xs']),
```

Comment:
Need to add 'md' here as well (though it doesn't really add styling except to give the default).